optuna/visualization/_param_importances.py

```diff
@@ -22,15 +22,15 @@
 
     from optuna.visualization._plotly_imports import go
 
-    Blues = plotly.colors.sequential.Blues
+    D3 = plotly.colors.qualitative.D3
 
     _distribution_colors = {
-        UniformDistribution: Blues[-1],
-        LogUniformDistribution: Blues[-1],
-        DiscreteUniformDistribution: Blues[-1],
-        IntUniformDistribution: Blues[-2],
-        IntLogUniformDistribution: Blues[-2],
-        CategoricalDistribution: Blues[-4],
+        UniformDistribution: D3[0],
+        LogUniformDistribution: D3[1],
+        DiscreteUniformDistribution: D3[2],
+        IntUniformDistribution: D3[3],
+        IntLogUniformDistribution: D3[4],
+        CategoricalDistribution: D3[5],
```

Comment:
Just want to clarify but do you think it's better to have different colors for different distributions? I am a bit skeptical since the distribution classes are not directly accessed by users and this change introduces a lot of information into the plots. Actually, colors are not the most important aspects in these figures.
![newplot (27)](https://user-images.githubusercontent.com/5983694/98183843-65279300-1f4c-11eb-9857-415b7ce1f264.png)

Reply:
Thank you for your comment! I understand your points and think there're 3 options to improve the current implementation and this PR:

1. Use single colour (not colourmap) for all the distributions both in Plotly and Matplotlib backends.
- Replace the blue colourmap with a colour "blue" in current implementation.
- Close this PR.

2. Use single colour in Plotly backend while different colours with legends in Matplotlib backend.
- Replace the blue colourmap with a colour "blue" in current implementation of Plotly.
- Keep changes in this PR only for Matplotlib.

3. Use different colurs with legends both in Plotly and Matplolib backends.
- Keep all the changes in this PR.
- Add legends to Plotly.

If the distribution classes are not directly referenced by users, it would confuse them to use several colours (even in the same colourmap) without any legend.

Reply:
Then, it might be better to just stick to a single color, no legends. For plotly, I believe we can still keep the mouse hover helpers, just as a bonus. Cc. @himkt

Reply:
Seems like it's been labeled stale. Are there any further comments on the coloring?

Reply:
So sorry for my slow reply! Let me check this weekend.

Reply:
I haven't quite understood how hatches and more distinguishable colors would differ in practice. A concrete example might help. By the way, https://github.com/optuna/optuna/issues/2080 might help resolving this discussion.

Reply:
I believe that colourblindness should be considered. [The new colour map in this PR](https://github.com/optuna/optuna/pull/1991#discussion_r517717525) couldn't be distinguished between red and green when I used [this simulation](https://www.color-blindness.com/coblis-color-blindness-simulator/). Please see also a suggestion in [the docs of matplotlib](https://matplotlib.org/stable/tutorials/colors/colormaps.html#color-vision-deficiencies) -- it recommends not using red and green at the same time.

I also test the colour maps in #2028 a little bit. They look well.